src/components/KTableData/KTableData.vue

```diff
@@ -0,0 +1,714 @@
+    :hide-pagination="hidePagination || !showPagination"
+    :hide-pagination-when-optional="hidePaginationWhenOptional"
+    :loading="loading || isTableLoading"
+    :max-height="maxHeight"
+    :nested="nested"
+    :pagination-attributes="tablePaginationAttributes"
+    :resize-columns="resizeColumns"
+    :row-attrs="rowAttrs"
+    :row-bulk-action-enabled="rowBulkActionEnabled"
+    :row-expandable="rowExpandable"
+    :row-hover="rowHover"
+    :row-link="rowLink"
+    :table-preferences="tablePreferences"
+    @get-next-offset="getNextOffsetHandler"
+    @get-previous-offset="getPreviousOffsetHandler"
+    @page-change="pageChangeHandler"
+    @page-size-change="pageSizeChangeHandler"
+    @sort="sortHandler"
+    @update:table-preferences="tableViewPreferences = $event"
+  >
+    <template
+      v-if="$slots.toolbar && !hideToolbar"
+      #toolbar
+    >
+      <slot
+        name="toolbar"
+        :state="stateData"
+      />
+    </template>
+
+    <template
+      v-if="$slots['bulk-actions']"
+      #bulk-actions="{ selectedRows }"
+    >
+      <slot
+        name="bulk-actions"
+        :selected-rows="selectedRows"
+      />
+    </template>
+
+    <template
+      v-if="$slots['bulk-action-items']"
+      #bulk-action-items="{ selectedRows }"
+    >
+      <slot
+        name="bulk-action-items"
+        :selected-rows="selectedRows"
+      />
+    </template>
+
+    <template
+      v-if="$slots['error-state']"
+      #error-state
+    >
+      <slot name="error-state" />
+    </template>
+
+    <template
+      v-if="$slots['empty-state']"
+      #empty-state
+    >
+      <slot name="empty-state" />
+    </template>
+
+    <template
+      v-if="$slots['empty-state-action-icon']"
+      #empty-state-action-icon
+    >
+      <slot name="empty-state-action-icon" />
+    </template>
+
+    <template
+      v-for="slot in getHeaderSlots"
+      :key="slot"
+      #[slot]="{ column }"
+    >
+      <slot
+        :column="column"
+        :name="slot"
+      />
+    </template>
+
+    <template
+      v-for="slot in getHeaderTooltipSlots"
+      :key="slot"
+      #[slot]="{ column }"
+    >
+      <slot
+        :column="column"
+        :name="slot"
+      />
+    </template>
+
+    <template
+      v-for="slot in getCellSlots"
+      :key="slot"
+      #[slot]="{ row, rowKey, rowValue }"
+    >
+      <slot
+        :name="slot"
+        :row="row"
+        :row-key="rowKey"
+        :row-value="rowValue"
+      />
+    </template>
+
+    <template
+      v-if="$slots['action-items']"
+      #action-items="{ row }"
+    >
+      <slot
+        name="action-items"
+        :row="row"
+      />
+    </template>
+
+    <template
+      v-if="$slots['row-expanded']"
+      #row-expanded="{ columnWidths, nestedHeaders, row }"
+    >
+      <slot
+        :column-widths="columnWidths"
+        name="row-expanded"
+        :nested-headers="nestedHeaders"
+        :row="row"
+      />
+    </template>
```

Comment:
So you'd have to have all expandable row content defined in a single slot, and then utilize `v-if` logic inside the slot (or something) to make it dynamic per row?

If that's the case, can you update the docs example in this PR accordingly to prove the concept?

Reply:
Another question on the KTableView implementation (since we're here).

The "getGeneric" function that returns the row object: why can't it accept an actual typescript generic and return the same type as to not lose the interface of the row?

Something like this:

```ts
const getGeneric = <TRow>(obj: TRow): TRow => obj
```

Reply:
> If that's the case, can you update the docs example in this PR accordingly to prove the concept?

Added an [example in docs](https://deploy-preview-2405--kongponents.netlify.app/components/table-view.html#row-expanded) for this.

Reply:
> Another question on the KTableView implementation (since we're here).

Implemented your suggestion, however I'm not sure it help as the passed object still is of type `any`.

Reply:
Is it getting set to any somewhere else? I know for the `row` it's always going to show `Record<string, any>` but for the column it should show that it follows the header interface?

Reply:
So I had to revert this change because it breaks usage in [public-ui-components](https://github.com/Kong/public-ui-components/actions/runs/11034925613/job/30649674841?pr=1654). We could try and fix it forward but I think that will make adoption of the new component too hard while adding very little benefit.